src/Accelerators/Accelerator.cpp

```diff
@@ -13,23 +13,35 @@
 //===----------------------------------------------------------------------===//
 
 #include "src/Accelerators/Accelerator.hpp"
-#include <iostream>
-#include <vector>
+#include "src/Accelerators/NNPA/NNPAAccelerator.hpp"
 
 namespace onnx_mlir {
 namespace accel {
 
-std::vector<Accelerator *> *Accelerator::acceleratorTargets;
-
-Accelerator::Accelerator() {
-  if (acceleratorTargets == NULL)
-    acceleratorTargets = new std::vector<Accelerator *>();
+llvm::SmallPtrSet<Accelerator *, 2> Accelerator::accelerators;
+
+void Accelerator::create(Accelerator::Kind kind,
+    mlir::OwningOpRef<mlir::ModuleOp> &module, mlir::MLIRContext &context,
+    mlir::PassManager &pm, onnx_mlir::EmissionTargetType emissionTarget) {
+  Accelerator *accel = nullptr;
+  switch (kind) {
+  case Kind::NNPA:
+#ifdef __NNPA__
```

Comment:
It might be more friendly to have separate llvm-unreachable under `#ifndef __NNPA__` indicating the onnx-mlir was not built for NNPA, as opposed to the more generic "should never be null"

Reply:
done